app/admin/RTMediaFormHandler.php

```diff
@@ -1153,20 +1156,42 @@ static function buddypress_render_options( $options ) {
 	public static function buddypress_content() {
 		global $rtmedia;
 
-		$general_group		= array();
-		$general_group[10]	= 'Integration With BuddyPress Features';
-		$general_group[660]	= 'Comment Media';
-		$general_group[50]	= 'Album Settings';
-		$general_group		= apply_filters( 'rtmedia_buddypress_setting_group', $general_group );
-
-		$render_options	= self::buddypress_render_options( $rtmedia->options );
+		$general_group      = array();
+		$general_group[10]  = 'Integration With BuddyPress Features';
+		$general_group[660] = 'Comment Media';
+		$general_group[50]  = 'Album Settings';
+		$general_group      = apply_filters( 'rtmedia_buddypress_setting_group', $general_group );
+
+		$render_options = self::buddypress_render_options( $rtmedia->options );
+		// Change option description when activity stream is disabled.
+		if ( ! bp_is_active( 'activity' ) ) {
+			$render_options['rtmedia-enable-on-activity']['args']['desc']      =
+			$render_options['rtmedia-activity-feed-limit']['args']['desc']     =
+			$render_options['rtmedia-enable-like-activity']['args']['desc']    =
+			$render_options['rtmedia-enable-comment-activity']['args']['desc'] =
+				'Please Enable BuddyPress Activity Streams to update option';
+		}
 		$render_options	= apply_filters( 'rtmedia_album_control_setting', $render_options, $rtmedia->options );
 
 		$render_options = apply_filters( 'rtmedia_buddypress_setting_options', $render_options );
 
 		self::render_tab_content( $render_options, $general_group, 10 );
 
 		do_action( 'rtmedia_buddypress_setting_content' );
+		/**
+		 * Disable inputs and change background color to differentiate disabled inputs.
+		 * If activity stream is disabled in BuddyPress Settings.
+		*/
+		if ( ! bp_is_active( 'activity' ) ) {
+			?>
+			<script>
+				jQuery( '#rtmedia-bp-enable-activity' ).prop( "disabled", true ).next().css( 'background-color', '#808080' );
+				jQuery( '#rtmedia-activity-feed-limit' ).prop( "disabled", true );
+				jQuery( '#rtmedia-enable-like-activity' ).prop( "disabled", true ).next().css( 'background-color', '#808080' );
+				jQuery( '#rtmedia-enable-comment-activity' ).prop( "disabled", true ).next().css( 'background-color', '#808080' );
```

Comment:
You can make this one lined
`jQuery( '#rtmedia-bp-enable-activity, #rtmedia-enable-like-activity, #rtmedia-enable-comment-activity' ).prop( "disabled", true ).next().css( 'background-color', '#808080' );`

Sorry for not pointing out in previous review